### ISSUE TITLE
msg_show is shown twice, when buffer is saved to a file

### ISSUE THREAD

**Issue by ajukraine** [folke/noice.nvim#300]
**Describe the bug**
When I save buffer to a file, the message is shown twice. Also I've figured out, it only happens with Lua buffers that my custom "BufWritePost" auto command handlers (sourcing $MYVIMRC).

**Which version of Neovim are you using?**
NVIM v0.9.0-dev-1737+g23d8f5b87-dirty
Build type: Release
LuaJIT 2.1.0-beta3

**To Reproduce**
1. Create auto command for event "BufWritePost" and file pattern (let's say "*.lua"), that sources $MYVIMRC
    ```lua
    vim.api.nvim_create_autocmd("BufWritePost", {
      pattern = "*.lua",
      nested = true,
      command = "source $MYVIMRC",
    })
    ```
1. Open Lua file (ex, `test.lua`)
1. Modify buffer and try to save it with `:w`
1. Message `"test.lua" 160L, 4330B written` is shown twice by Noice notify window

**Expected Behavior**
The message should be shown only once

**Screenshots**
<img width="474" alt="image" src="https://user-images.githubusercontent.com/828545/210190188-c2953753-c5e1-469f-a837-194709846275.png">

<img width="567" alt="image" src="https://user-images.githubusercontent.com/828545/210190345-92e3c416-da77-433e-b1cb-87287036b714.png">

**Comment by gennaro-tedesco** [comment]
Does this also happen if you define the autocommand in an `augroup` and clear it before?

**Comment by folke** [comment]
you need to clear the group indeed. not related to noice at all

**Comment by ajukraine** [comment]
Yeah, I clear it in my real config. @gennaro-tedesco 

`local augroup_config = vim.api.nvim_create_augroup("config_reload", { clear = true })`

**Comment by ajukraine** [comment]
Without 'noice' it writes to messages only ones. I suspect it's somehow related to blocking operations and how redraw works

**Comment by ajukraine** [comment]
Also it only is shown once, if I print something within my Lua file.

<img width="469" alt="image" src="https://user-images.githubusercontent.com/828545/210281912-2eb4456f-af8f-467e-8d04-2c49c7aa3993.png">

**Comment by folke** [comment]
@Fausto-Korpsvart you mean that `<` message? Please check the docs, wiki, other issues. That's standard Neovim behavior

**Comment by Fausto-Korpsvart** [comment]
> @Fausto-Korpsvart you mean that `<` message? Please check the docs, wiki, other issues. That's standard Neovim behavior

@folke, yes, I mean the `File saved` message is not displayed and instead the `<` symbol is displayed, the problem happens when Noice is enabled, with a standard configuration where Neovim has a standard behaviour this doesn't happen, which makes me think that something in Noice is causing this, I know it's not something serious or urgent to solve compared to other problems Noice can have, I only noticed it because I don't use Notify to show this message, I use the `mini` integration of Noice, and suddenly I don't see the message when I save files and I see that symbol instead, and using a minimal configuration I noticed that it's Noice that causes this behaviour.

But well, I'm going to check the standard behaviour again to try to rule out that it's a Noice problem.

Thanks anyway.

**Comment by folke** [comment]
@Fausto-Korpsvart check `'shortmess` and `t` option.

**Comment by Fausto-Korpsvart** [comment]
> @Fausto-Korpsvart check `'shortmess` and `t` option.

You are right, @folke, I set these values to `shortmess = 'fFIlqx'` and that solves the problem, but why does Noice cause the output to be truncated on long strings? Because I didn't have `shortmess` configured before and Noice didn't have that problem, anyway, thanks.
 Oh, and thanks for `lazy.nvim`, Lazy and Noice completely changes the game in the use of Neovim, it gives me that feeling at least, great job you have done.

**Comment by yutkat** [comment]
ref: https://github.com/neovim/neovim/issues/21653

**Comment by samhed** [comment]
Is the referenced issue neovim/neovim#21653 relevant to the original issue? neovim/neovim#21653 looks to be included in Neovim v0.9.0, and the duplicate "written..." messages issue remains.

It seems, @Fausto-Korpsvart was having an unrelated issue.

What can I do to help diagnose or resolve the original issue?

**Comment by alifprihantoro** [comment]
![Screenshot_20231112-113209](https://github.com/folke/noice.nvim/assets/69776347/6d87e849-d9f4-4f98-a7b1-4b2f98d4d4a6)
If you like me, where a regular message along with an error like an image when written. In my case, it happened in **WichKey**. Where the code `:w<CR>` I changed to `function() vim.cmd('w) end`.
